dotfiles/README.md

```diff
@@ -11,10 +11,33 @@ tags: [helper]
 
 Allow developers to optionally bring their own [dotfiles repository](https://dotfiles.github.io)! Under the hood, this module uses the [coder dotfiles](https://coder.com/docs/v2/latest/dotfiles) command.
 
-```hcl
+```tf
 module "dotfiles" {
   source = "registry.coder.com/modules/dotfiles/coder"
   version = "1.0.0"
   agent_id = coder_agent.example.id
 }
 ```
+
+## Examples
+
+### Apply dotfiles to the current user only
+
+```tf
+module "dotfiles" {
+  source   = "registry.coder.com/modules/dotfiles/coder"
+  version  = "1.0.0"
+  agent_id = coder_agent.example.id
+}
+```
+
+### Apply dotfiles to the current user and root
+
+```tf
+module "dotfiles" {
+  source   = "registry.coder.com/modules/dotfiles/coder"
+  version  = "1.0.0"
+  agent_id = coder_agent.example.id
+  root     = true
```

Comment:
I feel this param/behavior is a bit weird, I would expect these dotfiles to _only_ to be applied to root.

I have two alternative proposals:
1. Rename this to `user = "root"`, and run with `sudo -u $USER ...` (change behavior to only apply to this one user)
2. Rename this to `extra_users = ["root"]`, this is more flexible and will cover more use-cases, and the "extra" makes it obvious that this will be applied to the standard user _and_ to root.

Reply:
I think using `user = ?` contrains you to have a module definition per user, which kinda sucks, which is also why I preferred for `root = true` to be extra, but I do agree, it is unclear.
I prefer the `extra_users` way and I have thought about it in the past, but I thought "who's gonna apply to more users than `coder` and `root`?".

I will update the PR to support an `extra_users` array.

Reply:
I don’t think the 1 definition/user constraint is bad per-se, is there a drawback I’m not thinking of?

But I’m fine with either approach.

Reply:
> I don’t think the 1 definition/user constraint is bad per-se, is there a drawback I’m not thinking of?

Well, I just don't like the idea that the template administrator will have to copy it again, I think that's bad UX.

```tf
module "dotfiles" {
  source = "git::https://github.com/phorcys420/coder-modules.git//dotfiles?ref=dotfiles-root"
  agent_id = coder_agent.dev.id
  extra_users = ["root"]
}
```

is cleaner to me than

```tf
module "dotfiles" {
  source = "git::https://github.com/phorcys420/coder-modules.git//dotfiles?ref=dotfiles-root"
  agent_id = coder_agent.dev.id
}

module "dotfiles-root" {
  source = "git::https://github.com/phorcys420/coder-modules.git//dotfiles?ref=dotfiles-root"
  agent_id = coder_agent.dev.id
  user = "root"
}
```

Reply:
@phorcys420 great insight, thanks for sharing.

I kind of like `user = "root"` because `extra_users` doesn't work with multiple module definitions. But I also don't like forcing two parameters on the user.

Is this possible?

```tf
module "dotfiles" {
  source = "git::https://github.com/phorcys420/coder-modules.git//dotfiles?ref=dotfiles-root"
  agent_id = coder_agent.dev.id
  # exports dotfiles_uri or similar
}

module "dotfiles-root" {
  source = "git::https://github.com/phorcys420/coder-modules.git//dotfiles?ref=dotfiles-root"
  dotfiles_uri = module.dotfiles.dotfiles_uri # when dotfiles_uri is defined, no parameter is created
  agent_id = coder_agent.dev.id
  user = "root"
}
```

Alternatively, the best option for now may be to just rename `root = true` to `also_apply_to_root = true`, or similar. That would convey the behavior better.

Reply:
I like this way better, with the optional `dotfiles_uri` and `user`, i'll play around with it :-)

Reply:
I have added a `dotfiles_uri` optional parameter that, if passed, hides the `coder_parameter` that asks the end user for their dotfiles URI. (873f057)

I will do the same thing for a `user` variable.